### ISSUE TITLE
getting invalidate key while validating nomad client file 

### ISSUE THREAD

**Issue by thatsk** [hashicorp/nomad#3639]
### Nomad version
Output from `nomad version`
Nomad v0.7.0

### Operating system and Environment details
centos 7.3.1611
### Issue
while validating config file it shows invalid key but it is correct.
nomad client file :(client.hcl)

datacenter = "nomad"
data_dir   = "/var/lib/nomad"

client {
  enabled = true
  options = {
   "docker.privileged.enabled" = "true"
   "docker.volumes.enabled" = "true"
    }
}

When I check with nomad validate it shows invalid key don't know why
nomad validate /etc/nomad.d/client.hcl
* invalid key: datacenter
* invalid key: data_dir
* invalid key: client

The same issue for server-side config also.
But my cluster is running ok.

**Comment by jcomeaux** [comment]
I believe you should just enclose everything in curly braces (the file needs to be json i think?)

{
  "client": {
    "enabled": true,
    "options": {
      "docker.auth.config": "/root/.docker/config.json"
    }
  },
  "data_dir": "/var/lib/nomad",
  "datacenter": “staging”,
  "server": {
    "bootstrap_expect": 3,
    "enabled": true,
    "encrypt": “myKey=="
  },
  "telemetry": {
    "collection_interval": "10s",
    "publish_allocation_metrics": true,
    "publish_node_metrics": true
  }
}

**Comment by jcomeaux** [comment]
I'm not hcl expert, but you might have more luck with json; here's my config that works:
```
{
  "client": {
    "enabled": true,
    "options": {
      "docker.auth.config": "/root/.docker/config.json"
    }
  },
  "data_dir": "/var/lib/nomad",
  "datacenter": “staging”,
  "server": {
    "bootstrap_expect": 3,
    "enabled": true,
    "encrypt": “myKey=="
  },
  "telemetry": {
    "collection_interval": "10s",
    "publish_allocation_metrics": true,
    "publish_node_metrics": true
  }
}
```

**Comment by thatsk** [comment]
@jcomeaux  Thanks for your respons.
But i have tried to same your json config also.

cat client.json
{
  "client": {
    "enabled": true,
    "options": {
      "docker.auth.config": "/root/.docker/config.json"
    }
  },
  "data_dir": "/var/lib/nomad",
  "datacenter": "staging",
  "server": {
    "bootstrap_expect": 3,
    "enabled": true,
    "encrypt": "myKey=="
  },
  "telemetry": {
    "collection_interval": "10s",
    "publish_allocation_metrics": true,
    "publish_node_metrics": true
  }
}

nomad validate client.json
Error getting job struct: Error parsing job file from client.json: 5 error(s) occurred:

* invalid key: client
* invalid key: data_dir
* invalid key: datacenter
* invalid key: server
* invalid key: telemetry

**Comment by jcomeaux** [comment]
@thatsk well, that client.json is exactly how i'm configured on nomad 0.7.0.

Just curious, have you tried pasting that json file in https://jsonlint.com/ or some other json linter?  My guess is there may be a problem with hidden characters or encoding?

**Comment by thatsk** [comment]
{
	"client": {
		"enabled": true,
		"options": {
			"docker.volumes.enabled": "true",
			"docker.privileged.enabled": "true"
		}
	},
	"data_dir": "/var/lib/nomad",
	"datacenter": "nomad"

}


I have checked on Json lint. Adding screenshot.
same error.

Error getting job struct: Error parsing job file from client.json: 3 error(s) occurred:

* invalid key: client
* invalid key: data_dir
* invalid key: datacenter


![valid](https://user-images.githubusercontent.com/15324013/33818081-00ede9ce-de69-11e7-9c51-98ef2732eb3b.png)

**Comment by schmichael** [comment]
Hi @thatsk,

The `nomad validate` command is for validating job files, not client and server configurations. Sorry for the confusion!

**Comment by aantono** [comment]
I'm actually seeing a similar issue, when using 
```
client {
  options {
    "docker.auth.config" = "/opt/nomad/config/docker-auth.json"
  }
}
```  
stanza, I'm getting an error:
```
==> Error loading configuration from /opt/nomad/config: Error loading /opt/nomad/config/docker-auth.json: error parsing 'config': 1 error(s) occurred:

* config: invalid key: auths
```

This is using `Nomad v0.7.1 (0b295d399d00199cfab4621566babd25987ba06e)`

**Comment by dadgar** [comment]
@aantono Based on that error it sounds like you have a key named "auths" in your clients HCL config which is invalid.

**Comment by aantono** [comment]
@dadgar After an hour of debugging, found the issue: I have placed `docker-auth.json` file into `/opt/nomad/config` directory, along with all my other nomad `*.hcl` config files, so the directory config globbing was pulling `docker-auth.json` into the main config along with all the other files.  I had to move it to another directory, and things started to work.  Somehow I just assumed that only `*.hcl` files will be globbed, but guess that's not the case and everything in the config dir is being consumed.
